frontend/public/components/edit-yaml.jsx

```diff
@@ -243,13 +255,51 @@ const EditYAMLInner = (props) => {
 
       const yaml = convertObjToYAMLString(obj);
       displayedVersion.current = _.get(obj, 'metadata.resourceVersion');
-      getEditor().setValue(yaml);
+      getEditor()?.setValue(yaml);
       setInitialized(true);
       setStale(false);
     },
     [convertObjToYAMLString, initialized, props.obj],
   );
 
+  const handleCodeReplace = (_event) => {
+    if (_event.target.id === 'confirm-replace') {
+      getEditor()?.setValue(olsCodeBlock?.value);
+    } else if (_event.target.id === 'keep-both') {
+      getEditor()?.setValue(appendYAMLString(olsCodeBlock?.value));
+    }
+    setShowReplaceCodeModal(false);
+  };
+
+  const isValidYaml = (str) => {
+    if (_.isEmpty(str)) {
+      return false;
+    }
+    try {
+      safeLoad(str);
+      return true;
+    } catch {
+      return false;
+    }
+  };
+
+  React.useEffect(() => {
+    if (!isValidYaml(olsCodeBlock?.value) || !isOLSRedirected) {
+      return;
+    }
+
+    const currentYAML = getEditor()?.getValue();
+
+    if (_.isEmpty(currentYAML) || currentYAML === olsCode) {
+      getEditor()?.setValue(olsCodeBlock?.value);
+    } else {
+      setShowReplaceCodeModal(true);
+    }
+
+    setOLSCode(olsCodeBlock?.value);
```

Comment:
Do you need to set this state in the same `useEffect`? Not adding exhaustive dependency list in the hook might lead to unindented issues and bugs later.

Reply:
I discovered that adding certain dependencies caused the feature to fail in some marginal cases. To ensure the feature's functionality, I included only the dependencies that were essential and excluded those that triggered unnecessary actions.